spec.bs

```diff
@@ -642,6 +703,8 @@ directly; for authentication the extension can only be accessed via
          (see [[#sctn-permissions-policy]]). We could additionally require and
          consume a [=transient activation=] here, if we felt the permission policy
          is not sufficient.
+         
+         Note: I feel we must still check that the ID's corresponds to the domain that was provided; you cannot use a key with Bank2 if Bank1 claims to be asking for it? The RP must be included in the signature at least.
```

Comment:
The RP is in the signature; https://w3c.github.io/secure-payment-confirmation/#dom-collectedclientadditionalpaymentdata-rp

Reply:
If the claimed RP is different than the RP that created it, why would the Browser/WebAuthn not also verify it?
First check that the RP is correct before asking the Authenticator to sign it?

Reply:
Sorry, I should have quoted to indicate the part I was replying to: "The RP must be included in the signature at least."

The caller of SPC makes no claim on the RP (it's not [an input to the API](https://w3c.github.io/secure-payment-confirmation/#sctn-securepaymentconfirmationrequest-dictionary)). They pass in a list of credential IDs, which do not contain RP information. The browser/authenticator* then matches the credential IDs to the underlying RP, which is what is signed.

(One thing the spec doesn't deal with today which we should fix - what if you pass in credential IDs from multiple RPs, what happens. This is a silly edge-case that doesn't really matter, but we should spec it to just fail in a privacy-preserving way probably).

* Today via the browser-stored metadata from credential creation time, in the long-term this should be queryable from the authenticator.